File: ECE39500/exams/midterm-2/midterm-2/PartA/InitState.py
from CompilerState import CompilerState
from StateFactory import StateFactory
import sys

class InitState(CompilerState):
    INT_MIN = -sys.maxint - 1

    def __init__(self, compiler):
        self.compiler = compiler
        return
    
    def comma(self):
        print("Parsing InitState, expected command, found ,")
        self.compiler.state = StateFactory.getState("ErrorState")

    def value(self, s):
        self.compiler.state = StateFactory.getState(s)
        self.compiler.targetState = self.compiler.state
        if self.compiler.DEBUG:
            print(f"Init to {self.compiler.targetState} transition")

    def eol(self):
        print(f"Error in InitState, expected value, found eol")
        self.compiler.state = StateFactory.getState("ErrorState")

    def eof(self):
        print("Parsing InitState, expected value, found eof")
        self.compiler.state = StateFactory.getState("ErrorState")
    
    def __str__(self):
        return "InitState"

File: ECE39500/exams/midterm-2/PartA/src/InitState.java
public class InitState extends CompilerState {

   Compiler compiler = null; 

   public InitState(Compiler compiler) {
      this.compiler = compiler;
   }

   public void comma( ) { 
      System.out.println("Parsint InitState, expected command, found ,");
      compiler.state = StateFactory.getState("ErrorState");
   }

   public void value(String str) {
      compiler.state = StateFactory.getState(str);
      compiler.targetState = compiler.state;
      if (Compiler.DEBUG) {
         System.out.println("Init to "+compiler.targetState+" transition");
      }
   }

   public void eol( ) { 
      System.out.println("Error in InitState, expected value, found eol");
      // XXX  compiler.targetState.eol( );
      compiler.state = StateFactory.getState("ErrorState");
   }
      
   public void eof( ) {
      System.out.println("Error in addi, expected value, found eof");
      compiler.state = StateFactory.getState("ErrorState");
   }

   public String toString( ) {
      return "InitState";
   }
}
